File: CSharp/ImageProcessing/Lib/ComImageProcessingInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/// <summary>
/// 画像処理の設定管理のロジック
/// </summary>
public class ComImageProcessingInfo
{
    /// <summary>
    /// 現在の画像処理の名称
    /// </summary>
    public string CurrentImageProcessingName { set; get; }

    /// <summary>
    /// エッジ検出の設定
    /// </summary>
    public EdgeDetectionInfo EdgeDetectoinInfo { set; get; }

    /// <summary>
    /// グレースケールの設定
    /// </summary>
    public GrayScaleInfo GrayScaleInfo { set; get; }

    /// <summary>
    /// 2値化の設定
    /// </summary>
    public BinarizationInfo BinarizationInfo { set; get; }

    /// <summary>
    /// グレースケール2次微分の設定
    /// </summary>
    public GrayScale2DiffInfo GrayScale2DiffInfo { set; get; }

    /// <summary>
    /// 色反転の設定
    /// </summary>
    public ColorReversalInfo ColorReversalInfo { set; get; }

    /// <summary>
    /// コンストラクタ
    /// </summary>
    public ComImageProcessingInfo()
    {
        EdgeDetectoinInfo = new EdgeDetectionInfo();
        GrayScaleInfo = new GrayScaleInfo();
        BinarizationInfo = new BinarizationInfo();
        GrayScale2DiffInfo = new GrayScale2DiffInfo();
        ColorReversalInfo = new ColorReversalInfo();
    }

    /// <summary>
    /// デスクトラクタ
    /// </summary>
    ~ComImageProcessingInfo()
    {
    }
}

/// <summary>
/// エッジ検出の設定管理のロジック
/// </summary>
public class EdgeDetectionInfo
{
    /// <summary>
    /// コンストラクタ
    /// </summary>
    public EdgeDetectionInfo()
    {
    }

    /// <summary>
    /// デスクトラクタ
    /// </summary>
    ~EdgeDetectionInfo()
    {
    }
}

/// <summary>
/// グレースケールの設定管理のロジック
/// </summary>
public class GrayScaleInfo
{
    /// <summary>
    /// コンストラクタ
    /// </summary>
    public GrayScaleInfo()
    {
    }

    /// <summary>
    /// デスクトラクタ
    /// </summary>
    ~GrayScaleInfo()
    {
    }
}

/// <summary>
/// 2値化の設定管理のロジック
/// </summary>
public class BinarizationInfo
{
    /// <summary>
    /// 閾値
    /// </summary>
    public byte Thresh { set; get; }

    /// <summary>
    /// コンストラクタ
    /// </summary>
    public BinarizationInfo()
    {
    }

    /// <summary>
    /// デスクトラクタ
    /// </summary>
    ~BinarizationInfo()
    {
    }
}

/// <summary>
/// グレースケール2次微分の設定管理のロジック
/// </summary>
public class GrayScale2DiffInfo
{
    /// <summary>
    /// コンストラクタ
    /// </summary>
    public GrayScale2DiffInfo()
    {
    }

    /// <summary>
    /// デスクトラクタ
    /// </summary>
    ~GrayScale2DiffInfo()
    {
    }
}

/// <summary>
/// 色反転の設定管理のロジック
/// </summary>
public class ColorReversalInfo
{
    /// <summary>
    /// コンストラクタ
    /// </summary>
    public ColorReversalInfo()
    {
    }

    /// <summary>
    /// デスクトラクタ
    /// </summary>
    ~ColorReversalInfo()
    {
    }
}

File: Vb/ImageProcessing/Lib/ComImageProcessingInfo.vb
''' <summary>
''' 画像処理の設定管理のロジック
''' </summary>
Public Class ComImageProcessingInfo
    ''' <summary>
    ''' 現在の画像処理の名称
    ''' </summary>
    Public Property CurrentImageProcessingName() As String

    ''' <summary>
    ''' エッジ検出の設定
    ''' </summary>
    Public Property EdgeDetectoinInfo() As EdgeDetectionInfo

    ''' <summary>
    ''' グレースケールの設定
    ''' </summary>
    Public Property GrayScaleInfo() As GrayScaleInfo

    ''' <summary>
    ''' 2値化の設定
    ''' </summary>
    Public Property BinarizationInfo() As BinarizationInfo

    ''' <summary>
    ''' グレースケール2次微分の設定
    ''' </summary>
    Public Property GrayScale2DiffInfo() As GrayScale2DiffInfo

    ''' <summary>
    ''' 色反転の設定
    ''' </summary>
    Public Property ColorReversalInfo() As ColorReversalInfo

    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()
        EdgeDetectoinInfo = New EdgeDetectionInfo()
        GrayScaleInfo = New GrayScaleInfo()
        BinarizationInfo = New BinarizationInfo()
        GrayScale2DiffInfo = New GrayScale2DiffInfo()
        ColorReversalInfo = New ColorReversalInfo()
    End Sub

    ''' <summary>
    ''' デスクトラクタ
    ''' </summary>
    Protected Overrides Sub Finalize()
        MyBase.Finalize()
    End Sub
End Class

''' <summary>
''' エッジ検出の設定管理のロジック
''' </summary>
Public Class EdgeDetectionInfo
    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()
    End Sub

    ''' <summary>
    ''' デスクトラクタ
    ''' </summary>
    Protected Overrides Sub Finalize()
    End Sub
End Class

''' <summary>
''' グレースケールの設定管理のロジック
''' </summary>
Public Class GrayScaleInfo
    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()
    End Sub

    ''' <summary>
    ''' デスクトラクタ
    ''' </summary>
    Protected Overrides Sub Finalize()
    End Sub
End Class

''' <summary>
''' 2値化の設定管理のロジック
''' </summary>
Public Class BinarizationInfo
    ''' <summary>
    ''' 閾値
    ''' </summary>
    Public Property Thresh() As Byte

    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()
    End Sub

    ''' <summary>
    ''' デスクトラクタ
    ''' </summary>
    Protected Overrides Sub Finalize()
    End Sub
End Class

''' <summary>
''' グレースケール2次微分の設定管理のロジック
''' </summary>
Public Class GrayScale2DiffInfo
    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()
    End Sub

    ''' <summary>
    ''' デスクトラクタ
    ''' </summary>
    Protected Overrides Sub Finalize()
    End Sub
End Class

''' <summary>
''' 色反転の設定管理のロジック
''' </summary>
Public Class ColorReversalInfo
    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()
    End Sub

    ''' <summary>
    ''' デスクトラクタ
    ''' </summary>
    Protected Overrides Sub Finalize()
    End Sub
End Class